mlrun/db/httpdb.py

```diff
@@ -332,21 +332,18 @@ def _api_call(_params):
         first_page_params["page"] = 1
         first_page_params["page-size"] = config.httpdb.pagination.default_page_size
         response = _api_call(first_page_params)
-        page_token = response.json().get("pagination", {}).get("page-token")
-        if not page_token:
-            yield response
-            return
 
-        params_with_page_token = deepcopy(params) or {}
-        params_with_page_token["page-token"] = page_token
+        yield response
+        page_token = response.json().get("pagination", {}).get("page-token", None)
+
         while page_token:
-            yield response
             try:
-                response = _api_call(params_with_page_token)
+                response = _api_call({"page-token": page_token})
```

Comment:
I suggest to add a comment to why you ignore the params and take only the page token